cpp/src/stream_compaction/stream_compaction_common.hpp

```diff
@@ -15,24 +15,20 @@
  */
 #pragma once
 
-#include <cudf/hashing/detail/hash_allocator.cuh>
+#include <cudf/detail/cuco_helpers.hpp>
 #include <cudf/hashing/detail/helper_functions.cuh>
 #include <cudf/table/row_operators.cuh>
 #include <cudf/table/table_device_view.cuh>
 
-#include <rmm/mr/device/polymorphic_allocator.hpp>
-
 #include <cuco/static_map.cuh>
 
 #include <limits>
 
 namespace cudf {
 namespace detail {
 
-using hash_table_allocator_type = rmm::mr::stream_allocator_adaptor<default_allocator<char>>;
-
 using hash_map_type =
-  cuco::static_map<size_type, size_type, cuda::thread_scope_device, hash_table_allocator_type>;
```

Comment:
`hash_map_type` is redefined exactly the same as in `hash_reduce_by_row.cuh`. Can we move the declaration into `cuco_helpers.hpp`? Then probably we have to rename it as `cuco_helpers.cuh` since it will depend on cuco.

Reply:
Discussed personally and agreed that this can be future work.